### PR TITLE
fix(a11y): restore focus after popover close with color picker

### LINKED ISSUE #1266: [a11y] legend loses focus after closing its popover
**Describe the bug**
When opening the legend popover and then closing it, the focus is lost (see video below)

**Expected behaviour**
The focus should go back to the popover handle.

**Screenshots**
in Kibana:
https://user-images.githubusercontent.com/4283304/126983142-90d5e0d4-d9a1-4be0-a9a8-202f6648b804.mp4

in Storybook:
https://user-images.githubusercontent.com/4283304/126983155-9a389ef9-3567-4a09-86e4-d635d98ddb8a.mp4

**Version (please complete the following information):**
Chrome, master

**Kibana Additional context**
Found when checking https://github.com/elastic/kibana/pull/106374

### PR DESCRIPTION
**Pull request by rshen91**
## Summary

If navigating through a chart with a legend and color picker, when closing the color picker, focus will be returned to the color of the legend item. 

Jul-28-2021 08-32-19

TODO

- [x] restore focus after hitting the done button on the color picker after advice from https://github.com/elastic/eui/pull/5102

## Details
This is a bug fix for accessibility focus management. The old color picker included an EuiWrappingPopover and was changed to EuiPopover. The EuiPopover element from eui returns focus back to the anchor

## Issues
Fixes #1266 and #935

### CODE REVIEW

**Comment by rshen91** [comment]
Working on mocking up a non eui color picker to see if the focus() is able to work.

**Review by nickofthyme — COMMENTED** [review]
I don't think this in the right solution here because the `EuiPopover` is expected to be passed a child (aka button) to be rendered and clicked to open the popover. See the empty `<div>` below in the list of legend items.

image

The `EuiWrappingPopover` is different in that it latches onto a given button element. This is intentional on the charts side to control what people are able to render as legend items.

I know you tried a million things to get it to work with the `EuiWrappingPopover`, do you think there is some change that eui could make to allow this to work as is?

**Comment by rshen91** [comment]
Thanks for all your help! I reverted the changes I made in 12b52dc  to have the story use the euiWrappingPopover instead and opened issue elastic/eui#5015 for the focus management of the Done button issue.

**Comment by rshen91** [comment]
Will reopen when above linked issue is resolved

File: storybook/stories/legend/9_color_picker.tsx [hunk 1]
```diff
@@ -27,15 +27,15 @@ import { BARCHART_1Y1G } from '@elastic/charts/src/utils/data_samples/test_datas
...
 
 export const Example = () => {
   const [colors, setColors] = useState<Record<SeriesKey, Color | null>>({});
+
   const CustomColorPicker: LegendColorPicker = useMemo(
     () => ({ anchor, color, onClose, seriesIdentifiers, onChange }) => {
       const handleClose = () => {
         onClose();
-        onCloseAction();
+        requestAnimationFrame(() => anchor.focus());
```
Comment (nickofthyme) [inline storybook/stories/legend/9_color_picker.tsx:38]: Can we try doing this in `src/` instead of `stories/`?

Comment (rshen91) [reply storybook/stories/legend/9_color_picker.tsx]: Smart! 2b5878d for changes thanks! Looks like this works!

**Review by markov00 — APPROVED** [review]
Tested locally it works fine in Chrome and Firefox.
I've tested in also in Safari and I've noticed that the color icon is not focusable, could you please check and open an issue + fix for that?

**Review by nickofthyme — APPROVED** [review]

**Comment by rshen91** [comment]
Checked in safari to see what's up. It looks like the same functionality is working in below video.

https://user-images.githubusercontent.com/20343860/131554016-ff3f46de-a927-4fe7-b531-3c1e7807c3d8.mp4

**Pull request merged**